hydromt/components/spatial.py

```diff
@@ -246,20 +210,138 @@ def write(
         if not exists(base_name):
             makedirs(base_name, exist_ok=True)
 
-        if self.data is None:
+        if self.region is None:
             self._logger.info("No region data found. skipping writing...")
         else:
             self._logger.info(f"writing region data to {write_path}")
-            gdf = self.data.copy()
+            gdf = self.region.copy()
 
             if to_wgs84 and (
                 write_kwargs.get("driver") == "GeoJSON"
-                or str(rel_path).lower().endswith(".geojson")
+                or str(filename).lower().endswith(".geojson")
             ):
                 gdf = gdf.to_crs(4326)
 
             gdf.to_file(write_path, **write_kwargs)
 
+    def parse_region(
+        self,
+        region: dict,
+        *,
+        hydrography_fn: str,
+        basin_index_fn: str,
```

Comment:
shouldn't these be optional so we can use the default ones?

Reply:
The optional part happens in `create_region`. I really need the files in this function.